src/dashboard/trulens/dashboard/Leaderboard.py

```diff
@@ -1,222 +1,497 @@
+from st_aggrid.grid_options_builder import GridOptionsBuilder
 import streamlit as st
-from streamlit_extras.switch_page_button import switch_page
-from trulens.core import TruSession
-from trulens.core.database.legacy.migration import MIGRATION_UNKNOWN_STR
 from trulens.core.utils.text import format_quantity
-from trulens.dashboard.streamlit_utils import init_from_args
-from trulens.dashboard.ux import styles
+from trulens.dashboard.constants import COMPARE_PAGE_NAME as compare_page_name
+from trulens.dashboard.constants import LEADERBOARD_PAGE_NAME as page_name
+from trulens.dashboard.constants import PINNED_COL_NAME
+from trulens.dashboard.constants import RECORD_LIMIT
+from trulens.dashboard.constants import RECORDS_PAGE_NAME as records_page_name
+from trulens.dashboard.pages.Compare import MAX_COMPARATORS
+from trulens.dashboard.pages.Compare import MIN_COMPARATORS
+from trulens.dashboard.utils.dashboard_utils import ST_APP_NAME
+from trulens.dashboard.utils.dashboard_utils import add_query_param
+from trulens.dashboard.utils.dashboard_utils import get_app_versions
+from trulens.dashboard.utils.dashboard_utils import get_apps
+from trulens.dashboard.utils.dashboard_utils import get_feedback_defs
+from trulens.dashboard.utils.dashboard_utils import get_records_and_feedback
+from trulens.dashboard.utils.dashboard_utils import (
+    read_query_params_into_session_state,
+)
+from trulens.dashboard.utils.dashboard_utils import render_app_version_filters
+from trulens.dashboard.utils.dashboard_utils import render_sidebar
+from trulens.dashboard.utils.dashboard_utils import set_page_config
+from trulens.dashboard.utils.dashboard_utils import update_app_metadata
 from trulens.dashboard.ux.components import draw_metadata_and_tags
-from trulens.dashboard.ux.page_config import set_page_config
 from trulens.dashboard.ux.styles import CATEGORY
+from trulens.dashboard.ux.styles import Category
+from trulens.dashboard.ux.styles import aggrid_css
+from trulens.dashboard.ux.styles import cell_rules
+from trulens.dashboard.ux.styles import default_direction
+from trulens.dashboard.ux.styles import stmetricdelta_hidearrow
 
-# https://github.com/jerryjliu/llama_index/issues/7244:
-asyncio.set_event_loop(asyncio.new_event_loop())
+APP_COLS = ["app_version", "app_id", "app_name"]
+APP_AGG_COLS = ["Records", "Average Latency"]
 
-if __name__ == "__main__":
-    # If not imported, gets args from command line and creates Tru singleton
-    init_from_args()
 
+def init_page_state(app_name: str):
+    if st.session_state.get(f"{page_name}.initialized", False):
+        return
 
-def leaderboard():
-    """Render the leaderboard page."""
+    if app_name:
+        add_query_param(ST_APP_NAME, app_name)
 
-    set_page_config(page_title="Leaderboard")
+    read_query_params_into_session_state(
+        page_name=page_name,
+        transforms={
+            "metadata_to_front": lambda x: x == "True",
+            "show_pinned": lambda x: x == "True",
```

Comment:
perhaps change name to "only show pinned" since that what the toggle actually does?

Reply:
Or could keep the name and still show the other apps, just show the pinned at the top. This is my preference